docs/recipes/amp-alertmanager-terraform.md

```diff
@@ -0,0 +1,141 @@
+metric_count: (default=1) the amount of each type of metric to generate. The same amount of metrics is always generated per metric type.
+label_count: (default=1) the amount of labels per metric to generate.
+datapoint_count: (default=1) the number of data-points per metric to generate.
+
+### Enabling Metric collection using AWS Distro for Opentelemetry
+1. Fork and clone the sample app from the repository at aws-otel-community.
+Then run the following commands.
+
+```
+cd ./sample-apps/prometheus
+docker build . -t prometheus-sample-app:latest
+```
+2. Push this image to a registry such as Amazon ECR or DockerHub.
+3. Deploy the sample app in the cluster by copying this Kubernetes configuration and applying it. Change the image to the image that you just pushed by replacing {{PUBLIC_SAMPLE_APP_IMAGE}} in the prometheus-sample-app.yaml file.
+
+```
+curl https://raw.githubusercontent.com/aws-observability/aws-otel-collector/main/examples/eks/aws-prometheus/prometheus-sample-app.yaml -o prometheus-sample-app.yaml
+kubectl apply -f prometheus-sample-app.yaml
+```
+4. Start a default instance of the ADOT Collector. To do so, first enter the following command to pull the Kubernetes configuration for ADOT Collector.
+
+```
+curl https://raw.githubusercontent.com/aws-observability/aws-otel-collector/main/examples/eks/aws-prometheus/prometheus-daemonset.yaml -o prometheus-daemonset.yaml
+```
+Then edit the template file, substituting the remote_write endpoint for your Amazon Managed Service for Prometheus workspace for YOUR_ENDPOINT and your Region for YOUR_REGION. Use the remote_write endpoint that is displayed in the Amazon Managed Service for Prometheus console when you look at your workspace details.
+You'll also need to change YOUR_ACCOUNT_ID in the service account section of the Kubernetes configuration to your AWS account ID.
+In this recipe, the ADOT Collector configuration uses an annotation (scrape=true) to tell which target endpoints to scrape. This allows the ADOT Collector to distinguish the sample app endpoint from kube-system endpoints in your cluster. You can remove this from the re-label configurations if you want to scrape a different sample app.
+5. Enter the following command to deploy the ADOT collector.
+```
+kubectl apply -f eks-prometheus-daemonset.yaml
+```
+
+### Deploying the Terraform module to configure Amazon Managed service for Prometheus workspace, recording rules & alert manager
+
+Now, we will  provision a Amazon Managed service for Prometheus workspace and will define an alerting rule that causes the Alert Manager to send a notification if a certain condition (defined in expr) holds true for a specified time period (for). Code in the Terraform language is stored in plain text files with the .tf file extension. There is also a JSON-based variant of the language that is named with the .tf.json file extension.
```

Comment:
service -> Service

Reply:
Turn code fragments such expr and for into code using `expr` and `for`

Reply:
changed